odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,226 @@
+            'False' if not using GPU
+            `True` if using GPU provided that `impl` supports it
+        impl='niftyrec', optional
+            Implementation back-end for the transform. Supported back-ends:
+            NiftyRec Tomography Toolbox through NiftyPy a Python wrapper
+            for NiftyRec
+        """
+        impl, impl_in = str(impl).lower(), impl
+        if impl not in _SUPPORTED_IMPL:
+            raise ValueError('implementation {!r} not supported.'
+                             ''.format(impl_in))
+
+        self.geometry = geometry
+        det_nx = geometry.det_grid.shape[0]
+        det_ny = geometry.det_grid.shape[1]
+        startAngle = float(geometry.motion_grid.points()[0])
+        stopAngle = float(geometry.motion_grid.points()[-1]) * (180 / np.pi)
+        N_proj = len(geometry.motion_grid.points())
+
+        self._cameras = np.float32(np.linspace(startAngle, stopAngle, N_proj,
+                                   dtype=np.float32).reshape((N_proj, 1)))
+        self._use_gpu = use_gpu
+        self._attenuation = attenuation
+        self._psf = psf
+
+        lowleft_x = geometry.detector.params.corners()[0][0]
+        lowleft_y = geometry.detector.params.corners()[0][1]
+        upright_x = geometry.detector.params.corners()[-1][0]
+        upright_y = geometry.detector.params.corners()[-1][1]
+        ran = odl.uniform_discr([lowleft_x, lowleft_y, startAngle],
+                                [upright_x, upright_y, stopAngle],
+                                [det_nx, det_ny, N_proj])
+
+        super().__init__(dom, ran, linear=True)
+
+    def _call(self, x):
+        """Compute the discrete SPECT ptojection
+
+        Parameters
+        ----------
+        x : `DiscreteLpVector`
+           Element in the domain of the operator to be forward projected
+
+        Returns
+        -------
+        out : `DiscreteLpVector`
+            Returns an element in the projection space
+        """
+        if self._attenuation is None:
+            attenuation = None
+        else:
+            attenuation = np.float32(
+                np.asfortranarray(self._attenuation.asarray()))
+
+        if self._psf is None:
+            psf = None
+        else:
+            psf = np.float32(self._psf.asarray())
+
+        x = np.float32(np.asfortranarray(x.asarray()))
+        projection = SPECT_project_parallelholes(x, self._cameras,
+                                                 attenuation, psf,
+                                                 use_gpu=self._use_gpu)
+        return self.range.element(projection)
+
+    @property
+    def adjoint(self):
+        """The back-projector associated with this array."""
+        return SpectBackproject(ran=self.domain,
+                                geometry=self.geometry,
+                                attenuation=self._attenuation, psf=self._psf,
+                                use_gpu=self._use_gpu)
+
+
+class SpectBackproject(Operator):
```

Comment:
Dont really see any mention of inheritance there. For me the above design would be akin to doing (with reference to the python example)

```python
class Duck:
    def __init__(self, name):
        self.name = name
        
    def quack(self):
        print("Quack, quack!")


class Person(Duck):
    def quack(self):
        print("I'm Quackin'!")
```

Sure, it makes sense from a programming point, but it is also ambiguous. Why not change `Duck` to inherit from `Person`? What happens if I want to add properties to `Duck`, do I need to `del` them in `Person`? The "obvious" design would be to have a common point of inheritance

```python
class Named:
    def __init__(self, name):
        self.name = name

        
class Duck(Named):        
    def quack(self):
        print("Quack, quack!")

        
class Person(Named):
    def quack(self):
        print("I'm Quackin'!")
```

This would keep the dependencies separated and wouldnt feel as "arbitrary". 

I remain highly sceptical. Have you ever seen this type of design used in another library so I can study an example of it?